File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/SyncStatusMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.enums.SyncStatus
import com.rookmotion.rook_sdk_health_connect.data.proto.SyncStatusProto

fun SyncStatus.toSyncStatusProto(): SyncStatusProto {
    return when (this) {
        SyncStatus.SYNCED -> SyncStatusProto.SYNCED
        SyncStatus.RECORDS_NOT_FOUND -> SyncStatusProto.RECORDS_NOT_FOUND
    }
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/sync_status_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';

extension SyncStatusMappers on SyncStatusProto {
  SyncStatus toDomain() {
    return switch (this) {
      SyncStatusProto.SYNCED => SyncStatus.synced,
      _ => SyncStatus.recordsNotFound,
    };
  }
}
